std/src/interface/rgb25.rs

```diff
@@ -0,0 +1,214 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(unused_braces)]
+
+use std::fmt::Debug;
+
+use bp::bc::stl::bitcoin_stl;
+use strict_encoding::{StrictDumb, StrictEncode};
+use strict_types::stl::std_stl;
+use strict_types::{CompileError, LibBuilder, TypeLib};
+
+use super::{
+    AssignIface, GenesisIface, GlobalIface, Iface, OwnedIface, Req, TransitionIface, VerNo,
+};
+use crate::interface::{ArgSpec, ContractIface};
+use crate::stl::{rgb_contract_stl, Amount, ContractData, Details, Name, Precision, StandardTypes};
+
+pub const LIB_NAME_RGB25: &str = "RGB25";
+/// Strict types id for the library providing data types for RGB25 interface.
+pub const LIB_ID_RGB25: &str = "sleep_shine_mono_2h6qQe4h857LhYbe5AS9ohU4iXNtPQfNzfeMDHEPsGBr";
+
+const SUPPLY_MISMATCH: u8 = 1;
+const NON_EQUAL_AMOUNTS: u8 = 2;
+const INVALID_PROOF: u8 = 3;
+const INSUFFICIENT_RESERVES: u8 = 4;
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+#[derive(StrictType, StrictDumb, StrictEncode, StrictDecode)]
+#[strict_type(lib = LIB_NAME_RGB25, tags = repr, into_u8, try_from_u8)]
+#[repr(u8)]
+pub enum Error {
+    #[strict_type(dumb)]
+    SupplyMismatch = SUPPLY_MISMATCH,
+    NonEqualAmounts = NON_EQUAL_AMOUNTS,
+    InvalidProof = INVALID_PROOF,
+    InsufficientReserves = INSUFFICIENT_RESERVES,
+}
+
+fn _rgb25_stl() -> Result<TypeLib, CompileError> {
+    LibBuilder::new(libname!(LIB_NAME_RGB25), tiny_bset! {
+        std_stl().to_dependency(),
+        bitcoin_stl().to_dependency(),
+        rgb_contract_stl().to_dependency(),
+    })
+    .transpile::<Error>()
+    .compile()
+}
+
+/// Generates strict type library providing data types for RGB25 interface.
+pub fn rgb25_stl() -> TypeLib { _rgb25_stl().expect("invalid strict type RGB25 library") }
+
+pub fn rgb25() -> Iface {
+    let types = StandardTypes::with(rgb25_stl());
+
+    Iface {
+        version: VerNo::V1,
+        name: tn!("RGB25"),
+        global_state: tiny_bmap! {
+            fname!("name") => GlobalIface::required(types.get("RGBContract.Name")),
+            fname!("details") => GlobalIface::optional(types.get("RGBContract.Details")),
+            fname!("precision") => GlobalIface::required(types.get("RGBContract.Precision")),
+            fname!("data") => GlobalIface::required(types.get("RGBContract.ContractData")),
+            fname!("created") => GlobalIface::required(types.get("RGBContract.Timestamp")),
+            fname!("issuedSupply") => GlobalIface::required(types.get("RGBContract.Amount")),
+        },
+        assignments: tiny_bmap! {
+            fname!("assetOwner") => AssignIface::private(OwnedIface::Amount, Req::NoneOrMore),
```

Comment:
See my comment in LNPBP-25 standard
```suggestion
            fname!("assetOwner") => AssignIface::private(OwnedIface::Amount, Req::OneOrMore),
```